src/App.jsx

```diff
@@ -14,33 +17,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const sortGoods = (goods, params) => {
+  const copyGoods = [...goods];
+  const { sortedBy, isReversed } = params;
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (sortedBy !== '') {
+    copyGoods.sort((good1, good2) => {
+      switch (sortedBy) {
+        case 'byAlphabet':
+          return good1.localeCompare(good2);
+        case 'byLength':
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (isReversed) {
+    copyGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return copyGoods;
+};
+
+export const App = () => {
+  const [sortedBy, setSortedBy] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const [hasChanged, setHasChanged] = useState(false);
```

Comment:
```suggestion
  const hasChanged = Boolean(sortedBy || isReversed);
```
You don't need to store in state just check dynamically